### ISSUE TITLE
Bug: `jsx-no-leaked-render` faulty autofix 

### ISSUE THREAD

**Issue by zaicevas** [jsx-eslint/eslint-plugin-react#3323]
Version: 7.30.1
Config:
```js
'react/jsx-no-leaked-render': ['error', { validStrategies: ['coerce', 'ternary'] }],
```

Before fix:
```tsx
const MyComponent = () => {
  const items = []
  const breakpoint = { phones: true }

  return <div>{items.length > 0 && breakpoint.phones && <span />}</div>
}
```

Post fix:

```tsx
const MyComponent = () => {
  const items = []
  const breakpoint = { phones: true }

  return <div>{!!!!!!!!!!!!!!!!!!!!items.length > 0 && breakpoint.phones && <span />}</div>
}
```

Expected:

```tsx
const MyComponent = () => {
  const items = []
  const breakpoint = { phones: true }

  return <div>{items.length > 0 && !!breakpoint.phones && <span />}</div>
}
```

On top of that, the auto fix is quite slow.

**Comment by ljharb** [comment]
The exclamation points definitely seem like a bug that should be fixed, and i agree that the `>` comparison shouldn't be touched since it can only ever be a boolean.

The autofixer's performance is something that can be improved separately, but isn't as concerning.

**Comment by Belco90** [comment]
I identified the issue but I need more time to rework the fix functionality to solve it.

**Comment by hduprat** [comment]
@Belco90 I created a PR to fix the multiple `!` issue during autofix. Feel free to review it!

**Comment by Belco90** [comment]
> @Belco90 I created a PR to fix the multiple `!` issue during autofix. Feel free to review it!

That's amazing! I'll try to review it later today. Thanks.

**Comment by Jackman3005** [comment]
I'm having a similar experience:

## Original
```jsx
{!sourceQuestData && loadingSourceQuest && (
  <StyledInfoLine>
    <Icon name="info" size={40} />
    <ActivityIndicator size="small" />
  </StyledInfoLine>
)}
```

## Result
```jsx
{!!!!!!!!!!!!!!!!!!!sourceQuestData && loadingSourceQuest && (
  <StyledInfoLine>
    <Icon name="info" size={40} />
    <ActivityIndicator size="small" />
  </StyledInfoLine>
)}
```

Similar to the OP, the `!sourceQuestData` should not have been touched. `loadingSourceQuest` is a boolean already, so it does not need to be touched either.

## Adding to booleans
I also see it adding `!!` to variables that are already clearly a boolean:

```jsx
const showComponent = editMode || item.status === "COMPLETED";

  return (
    <>
      {!!showComponent && (...)}
    </>
  )
```

**Comment by ljharb** [comment]
This should be fixed by #3353.
